File: Problems/Parking Lot/Java/payments/PaymentStrategy.java
package payments;

import ticket.Ticket;

public class PaymentStrategy implements Payments {
    Payments paymentStrategy;

    public PaymentStrategy(Payments paymentStrategy) {
        this.paymentStrategy = paymentStrategy;
    }

    @Override
    public double pay(Ticket ticket) {
        return paymentStrategy.pay(ticket);
    }
    
}

File: Problems/Parking Lot/Golang/payments/payment_strategy.go
package payments

import "parking-lot/ticket"

type PaymentStrategy struct {
	paymentStrategy Payments
}

func NewPaymentStrategy(paymentStrategy Payments) *PaymentStrategy {
	return &PaymentStrategy{paymentStrategy: paymentStrategy}
}

func (ps *PaymentStrategy) Pay(ticket ticket.Ticket) float64 {
	return ps.paymentStrategy.pay(ticket)
}
